### PR TITLE
Fixed squid:S2142

### PR DESCRIPTION
**Pull request by warlockza**

### CODE REVIEW

File: src/test/java/org/cloudfoundry/promregator/cache/AutoRefreshingCacheMapTest.java [hunk 1]
```diff
@@ -133,6 +133,7 @@ public void run() {
 				Assert.assertEquals("labolg", value);
 			} catch (InterruptedException e) {
 				Assert.fail("Thread was interrupted");
+				Thread.currentThread().interrupt();
```
Comment (eaglerainbow) [inline src/test/java/org/cloudfoundry/promregator/cache/AutoRefreshingCacheMapTest.java]: Well, won't be reached, as `Assert.fail` never returns. 
But Sonar doesn't know this...

Shall I put a "false positive" to SonarCloud on these cases?

File: src/test/java/org/cloudfoundry/promregator/mockServer/DefaultMetricsEndpointHttpHandler.java [hunk 2]
```diff
@@ -29,6 +29,7 @@ public void handle(HttpExchange he) throws IOException {
 				Thread.sleep(this.delayInMillis);
 			} catch (InterruptedException e) {
 				Assert.fail("Unexpected interruption of mocking environment");
+				Thread.currentThread().interrupt();
```
Comment (eaglerainbow) [inline src/test/java/org/cloudfoundry/promregator/mockServer/DefaultMetricsEndpointHttpHandler.java]: see above

File: src/test/java/org/cloudfoundry/promregator/cache/AutoRefreshingCacheMapTest.java [hunk 3]
```diff
@@ -200,6 +201,7 @@ public void testLockObjectStringWorks() throws InterruptedException {
 				Thread.sleep(200);
 			} catch (InterruptedException e) {
 				Assert.fail("Interrupted");
+				Thread.currentThread().interrupt();
```
Comment (eaglerainbow) [inline src/test/java/org/cloudfoundry/promregator/cache/AutoRefreshingCacheMapTest.java]: see above

File: src/test/java/org/cloudfoundry/promregator/cache/AutoRefreshingCacheMapTest.java [hunk 4]
```diff
@@ -308,6 +310,7 @@ public void testLockObjectCompositeWorks() throws InterruptedException {
...
```
Comment (eaglerainbow) [inline src/test/java/org/cloudfoundry/promregator/cache/AutoRefreshingCacheMapTest.java]: see above

File: src/main/java/org/cloudfoundry/promregator/scanner/ReactiveTargetResolver.java [hunk 5]
```diff
@@ -423,7 +423,7 @@ private boolean isApplicationInScrapableState(String state) {
 		}
 		
 		/* TODO: To be enhanced, once we know of further states, which are
-		 * also scrapable.
+		 * also scrapeable.
```
Comment (eaglerainbow) [inline src/main/java/org/cloudfoundry/promregator/scanner/ReactiveTargetResolver.java]: a nice one :) 😁

File: src/test/java/org/cloudfoundry/promregator/cache/AutoRefreshingCacheMapTest.java (hunk 1 above)
Comment (warlockza) [reply src/test/java/org/cloudfoundry/promregator/cache/AutoRefreshingCacheMapTest.java]: Should we just take out the assert.fail and interrupt?

Comment (warlockza) [reply src/test/java/org/cloudfoundry/promregator/cache/AutoRefreshingCacheMapTest.java]: Have put the Interrupt before the Assert, that way , we are still being good Thread Citizens by not dropping the interrupted status, and still allowing our assert to occur.

Comment (eaglerainbow) [reply src/test/java/org/cloudfoundry/promregator/cache/AutoRefreshingCacheMapTest.java]: That's a good compromise

File: src/test/java/org/cloudfoundry/promregator/cache/AutoRefreshingCacheMapTest.java [hunk 6]
```diff
@@ -132,7 +132,8 @@ public void run() {
 				value = subject.get("global");
 				Assert.assertEquals("labolg", value);
 			} catch (InterruptedException e) {
-				Assert.fail("Thread was interrupted");
+				Thread.currentThread().interrupt();
+                Assert.fail("Thread was interrupted");
```
Comment (eaglerainbow) [inline src/test/java/org/cloudfoundry/promregator/cache/AutoRefreshingCacheMapTest.java]: Please check indention.

Comment (warlockza) [reply src/test/java/org/cloudfoundry/promregator/cache/AutoRefreshingCacheMapTest.java]: Fixed now - Please confirm

Comment (warlockza) [reply src/test/java/org/cloudfoundry/promregator/cache/AutoRefreshingCacheMapTest.java]: Can this be merged?

**Pull request merged**